src/components/smart-trade-button/index.tsx

```diff
@@ -148,7 +149,7 @@ export function SmartTradeButton(props: SmartTradeButtonProps) {
     if (buttonState === TradeButtonState.mismatchingQueryNetwork) {
       if (queryNetwork) {
         switchChain({ chainId: queryNetwork })
-        router.replace('/leverage')
+        router.replace(pathname)
```

Comment:
This will probably happen before the user even switched the chain?

Reply:
since the switch chain here is synchronous, it wont. There is another method called switchChainAsync exported from switch chain. If I used that and wouldnt wait for it then the scenario would be as you have described